src/solvers/flattening/boolbv_array_of.cpp

```diff
@@ -25,7 +25,14 @@ bvt boolbvt::convert_array_of(const array_of_exprt &expr)
   std::size_t width=boolbv_width(array_type);
 
   if(width==0)
-    return conversion_failed(expr);
+  {
+    // A zero-length array is acceptable;
+    // an element with unknown size is not.
+    if(boolbv_width(array_type.subtype())==0)
```

Comment:
What about arrays of arrays of zero length? Would Java permit that? If so, this check needs to be amended.